### ISSUE TITLE
CORS Error : Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request

### ISSUE THREAD

**Issue by aggarwalaman1191** [emissary-ingress/emissary#4496]
I have one frontend app lets say app.com and then trying to access an api through that api.com/api-prefix/. Both app and api have an OAuth2 filter on top of with Azure AD authentication along with the JWT filter. Now when I log inside the app, I get this CORS error

Access to XMLHttpRequest at 'https://api.com/api-prefix/' from origin 'https://app.com/' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.

Whereas if I change my host to api.com instead of app.com it works fine.

So basically just to sum up, api.com to api.com/api-prefix/ works, but app.com to api.com/api-prefix/ gives CORS error, all of these three origins are behind azure ad filter and using same configuration pretty much.

I have tried a number of suggestions but none seem to have helped, so posting there if someone can guide.

Ambassador Version : 1.9.0

**Comment by cindymullins-dw** [comment]
Hi @aggarwalaman1191 , can you add your Mapping and Host resources here so we can confirm your CORS enabled settings.  We'd also advise upgrading to 1.14.3 if you can.

**Comment by aggarwalaman1191** [comment]
Below is the host config, it is same for api.com
```
apiVersion: getambassador.io/v2
kind: Host
metadata:
  name: host-app-com
  namespace: ambassador
spec:
  acmeProvider:
    authority: 
    email:
  hostname: app.com
  requestPolicy:
    insecure:
      action: Redirect
```

Mapping for API
```
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: api-com-api-prefix
  namespace: ambassador
spec:
  add_request_headers:
    X-Forwarded-Host: api.com
    X-Forwarded-Prefix: /api-prefix/
  host: api.com
  prefix: /api-prefix/
  prefix_regex: false
  rewrite: /
  service: api-com-prefix-service
  timeout_ms: 30000
```
  
Mapping for frontend app
```
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: mapping-app-com
  namespace: ambassador
spec:
  add_request_headers:
    X-Forwarded-Host: app.com
    X-Forwarded-Prefix: /
  host: app.com
  prefix: /
  prefix_regex: false
  rewrite: /
  service: app-com-service-frontend-app
  timeout_ms: 30000
```

**Comment by cindymullins-dw** [comment]
So you can set your [CORS attribute](https://www.getambassador.io/docs/emissary/latest/topics/using/cors/#the-cors-attribute) either in a Mapping where you want it to apply, or in a Module in which case it would apply globally.  There's some specs you can set there including the origin you want to allow.  Can you try following [this example](https://www.getambassador.io/docs/emissary/latest/topics/using/cors/#example) adding the CORS attribute to your Mapping?

**Comment by aggarwalaman1191** [comment]
@cindymullins-dw I tried this but didn't help, ended up with the same error.
```
cors:
    credentials: true
    exposed_headers: X-Custom-Header
    headers: Content-Type
    max_age: "86400"
    methods: POST, GET, OPTIONS
    origins: https://app.com,https://api.com
```

**Comment by aggarwalaman1191** [comment]
Hi @cindymullins-dw Is there any other config that I can try apart from this? This doesn't seem to work as expected.

**Comment by cindymullins-dw** [comment]
Did you try [upgrading to 1.14](https://www.getambassador.io/docs/emissary/1.14/tutorials/getting-started/)?  We'd suggest upgrading and we can troubleshoot further from there if it's still not working.

**Comment by aggarwalaman1191** [comment]
@cindymullins-dw I have upgraded the ambassador but that didn't help either.

**Comment by cindymullins-dw** [comment]
HI @aggarwalaman1191 , support reminded me today that we fixed a similar issue in 2.x so it may be lingering in 1.x as well.  Please see the 3rd item in these the [2.3.2 release notes](https://www.getambassador.io/docs/edge-stack/latest/release-notes/#2.3.2).  You'd need to follow this doc to [migrate to 2.4](https://www.getambassador.io/docs/emissary/latest/topics/install/upgrade/yaml/emissary-1.14/emissary-2.4/#migration-overview) and set up a Listener.
